atlp-rwanda/hackers-ec-be
# #187419042 Set up Tests locally & implement Continuous Integration using GitHub action

#### What does this PR do?

This Pull Request sets up tests locally and implements continuous integration using GitHub Actions. It’s a crucial step for maintaining code quality using code climate and ensuring that our project remains robust.

####  Description of Task to be completed?
In this Pull Request, I’ve configured the test suite to run locally and create GitHub action for CI

####  How should this be manually tested?

To verify the changes introduced by this PR:

1. Clone the repository and navigate to the project directory.
2. Run npm install to install the necessary dependencies.
3. Execute the following scripts:
     a. npm run test: This command runs the test suite.
     b. npm run test-test: Use this script to run tests in watch mode, allowing you to monitor changes.
     c. npm run test:ci: This script runs tests as part of GitHub Actions, generating coverage reports.
     
#### What are the relevant pivotal tracker/Trello stories?

 #187419042
